src/components/LoaderButton.js

```diff
@@ -1,20 +1,40 @@
-import Button from "react-bootstrap/lib/Button";
+import { Button } from "@mui/material";
+import { makeStyles } from "@mui/styles";
 import Glyphicon from "react-bootstrap/lib/Glyphicon";
 
-export default ({
-  isLoading,
+const useStyles = makeStyles((theme) => ({
+  root: {
+    "& .MuiButtonBase-root": {
+      marginTop: theme.spacing(1),
+      fontSize: 16,
+    },
+  },
+}));
+
+export default function LoaderButton({
   text,
   loadingText,
+  isLoading,
+  disabled,
   className = "",
-  disabled = false,
-  ...props
-}) => (
```

Comment:
Leaving this as a comment since direct changes are disabled! After you've defined the LoaderButtonProps interface, add it to the function's props object, on line 11. IE: 

`}: LoaderButtonProps) => (`

If you've converted this file to .tsx, added an interface defining the button props, and defined that as the type of the LoaderButton function argument, those compilation errors you're seeing should go away. (Though you will probably get another error about the .spacing property on the default theme ... LMK if you need suggestions on resolving that too!)

Reply:
I did this :)

Reply:
I figured out the pure JS way to avoid the Type error. 🤦  

Allow type, color, and variant to possibly be undefined in the props, and then INSIDE the function set them to their defaults if they're undefined. 

```
export default function LoaderButton({
  text,
  loadingText,
  isLoading,
  disabled,
  className = "",
  type,
  color,
  variant,
}) {
  type = type || "button";
  color = color || "primary";
  type = type || "contained";

```

Reply:
@jayeclark I would have never thought of that lol